promql/functions.go

```diff
@@ -716,13 +723,35 @@ func funcAvgOverTime(vals []parser.Value, args parser.Expressions, enh *EvalNode
 		return vec, nil
 	}
 	return aggrOverTime(vals, enh, func(s Series) float64 {
-		var mean, kahanC float64
-		for i, f := range s.Floats {
+		var (
+			sum             = s.Floats[0].F
+			mean, kahanC    float64
+			incrementalMean bool
+		)
+		for i, f := range s.Floats[1:] {
 			count := float64(i + 1)
```

Comment:
It was really hard to reconcile the algorithm in `avg` and this `avg_over_time`.
```suggestion
			count := float64(i + 2)
```
In avg_over_time we create groups with count==1 and the first number added makes it count==2. 

Here with this change we'd consider the 0th element in sum as count==1 and first element from the rest of the list as count==2. And also this way the `count` at the end as calculated by `len(s.Floats)` would mean the same thing.

Reply:
Not sure if I fully understand. Could you check the commit I have just pushed to see if it is what you are fishing for?